src/liboslexec/batched_llvm_instance.cpp

```diff
@@ -1813,10 +1813,10 @@ BatchedBackendLLVM::build_llvm_init()
         FOREACH_PARAM(Symbol & sym, gi)
         {
             if (sym.typespec().is_closure_based()) {
+                sym.make_varying();
+
                 int arraylen     = std::max(1, sym.typespec().arraylength());
-                llvm::Value* val = ll.constant_ptr(NULL, ll.type_void_ptr());
```

Comment:
So the original code here should have sufficed.  maybe could just put 
OSL_ASSERT(sym.is_varying());

Which will probably fire? (let me know if it does).   If the assert does fire that means there is a bug in BatchedAnalysis, which gets run as part of the runtimeoptimizer.  BatchedAnalysis is responsible for calling symbol::make_varying() not sure anyone else should be calling it (maybe it should be made private and friend BatchedAnalysis).

Even though this allows the unit test to pass, calling mark_varying() outside of BatchedAnalysis means that any operations that use that symbol downstream will notice that their input was varying and therefore so should their result be varying.

If you add at the top of batched_analysis.cpp
#define OSL_DEV 1
it will dump the results of batched_analysis and can see more of what's going on.

Reply:
I've switched the build_llvm_init changes to just this assert - it is indeed triggering.

I again, don't know anything about BatchedAnalysis, but as a placeholder, I added a quick commit where I just force it to tag all closures as varying - this seems to resolve the assert in build_llvm_init and get these test passing again, so that supports the idea that BatchedAnalysis is the right place to fix this.